test/evergreen/print_stack_trace.py

```diff
@@ -108,50 +108,68 @@ def dump(self, exe_path: str, core_path: str, lib_path: str, dump_all: bool, out
 
 def main():
     parser = argparse.ArgumentParser()
-    parser.add_argument('-e', '--executable_path',
-                        help='path to the executable',
-                        required=True)
     parser.add_argument('-c', '--core_path',
-                        help='directory path to the core dumps',
-                        required=True)
+                        help='directory path to the core dumps')
+    parser.add_argument('-e', '--executable_path',
+                        help='path to the executable')
     parser.add_argument('-l', '--lib_path', help='library path')
-    group = parser.add_mutually_exclusive_group(required=True)
-    group.add_argument('--unit_test', action='store_true', help='Format core dumps from python unit tests')
-    group.add_argument('--format', action='store_true', help='Format core dumps from format tests')
+    parser.add_argument('-p', '--python_binary_path',
+                        help='python binary path')
     args = parser.parse_args()
 
     # Store the path of the core files as a list.
     core_files = []
-    dump_all = args.unit_test
-
-    regex = None
-    if (args.format):
-        regex = re.compile(r'dump_t.*core', re.IGNORECASE)
-    elif (args.unit_test):
-        regex = re.compile(r'dump.*python.*core', re.IGNORECASE)
+    regex = re.compile(r'dump.*core', re.IGNORECASE)
+    core_path = args.core_path
+    if core_path is None:
+        core_path = '.'
 
-    for root, _, files in os.walk(args.core_path):
+    for root, _, files in os.walk(core_path):
         for file in files:
             if regex.match(file):
                 core_files.append(os.path.join(root, file))
 
     for core_file_path in core_files:
         print(border_msg(core_file_path), flush=True)
+
+        # If not specified, derive the binary name from the core dump folder.
+        # The test/format and cpp suites are different and need specific handling. All the other
+        # tests have an executable that follows the pattern "test_<test_folder>". For example,
+        # the test folder wt2853_perf contains an executable named test_wt2853_perf.
+        if args.executable_path is None:
+            # Retrieve the first part of the path without the core name.
+            dirname = core_file_path.rsplit('/', 1)[0]
+            if dirname.startswith('WT_TEST/test_'):
+                # For core dumps generate by Python tests, we need the executable to be set.
+                if args.python_binary_path is None:
+                    print("The Python binary is not specified!")
```

Comment:
Should this be a noisier failure? (e.g. one or more of printing to stderr, killing the script, or returning non-zero). Looking at the commit history, it seems we should always specify a python binary, so we want to know ASAP if this isn't set up right.

Reply:
I am not sure this is the right approach though, if there are other core files, we don't want to miss them either.

Reply:
Can we to combine the args.executable_path and args.python_binary_path in this case or does it complicate things?

Reply:
Thanks, @jiechenbo, awesome idea! I did not know it was a thing! But `sys.executable` does the trick. See a8d13930e1a73bdf8766156f97a853faf34794e0 for the changes.